fvm/src/kernel/default.rs

```diff
@@ -1006,113 +1006,93 @@ impl<C> EventOps for DefaultKernel<C>
 where
     C: CallManager,
 {
-    fn emit_event(&mut self, raw_evt: &[u8], raw_key: &[u8], raw_val: &[u8]) -> Result<()> {
-        const MAX_NR_ENTRIES: usize = 256;
-        const MAX_KEY_LEN: usize = 32;
+    fn emit_event(
+        &mut self,
+        event_headers: &[fvm_shared::sys::EventEntry],
+        event_keys: &[u8],
+        event_values: &[u8],
+    ) -> Result<()> {
+        const MAX_NR_ENTRIES: usize = 255;
+        const MAX_KEY_LEN: usize = 31;
         const MAX_VALUE_LEN: usize = 8 << 10;
 
         if self.read_only {
             return Err(syscall_error!(ReadOnly; "cannot emit events while read-only").into());
         }
 
-        if raw_val.len() > MAX_VALUE_LEN {
```

Comment:
Removed this check as we allow _each_ value to be 8kib.

Reply:
oh, that is interesting, it has up to now always been 8kib in total over _all_ values. Should we remove this check ([relevant code](https://github.com/filecoin-project/ref-fvm/blob/master/fvm/src/kernel/default.rs#L1086-L1090)) in master as well?

Reply:
Uh. No, no, you were right. I'm not sure what I was looking at (and I wrote the code + spec so I'm not sure what I was thinking either...).

Reply:
Fixed and I've:

1. Renamed the constant so I don't mess this up again.
2. Added a comment to the second check to explain why we have two.